silhouette-password-bcrypt/src/test/scala/com/mohiva/play/silhouette/password/BCryptPasswordHasherSpec.scala

```diff
@@ -17,56 +17,91 @@ package com.mohiva.play.silhouette.password
 
 /**
  * Test case for the [[BCryptPasswordHasher]] class.
  */
 class BCryptPasswordHasherSpec extends Specification {
 
   "The `hash` method" should {
-    "hash a password" in {
-      val password = "my_S3cr3t_p@sswQrd"
-      val hasher = new BCryptPasswordHasher
-      val info = hasher.hash(password)
-
-      info must beAnInstanceOf[PasswordInfo]
-      info.hasher must be equalTo BCryptPasswordHasher.ID
-      info.password must not be equalTo(password)
-      info.salt must beNone
+    "hash a password" in new Context {
+      val passwordInfo: PasswordInfo = hasher.hash(password)
+
+      passwordInfo.hasher must be equalTo BCryptPasswordHasher.ID
+      passwordInfo.password must not be equalTo(password)
+      passwordInfo.salt must beNone
     }
   }
 
   "The `matches` method" should {
-    "return true if a password matches a previous hashed password" in {
-      val password = "my_S3cr3t_p@sswQrd"
-      val hasher = new BCryptPasswordHasher
-      val info = hasher.hash(password)
+    "return true if a password matches a previous hashed password" in new Context {
+      val passwordInfo = hasher.hash(password)
+
+      hasher.matches(passwordInfo, password) must beTrue
+    }
+
+    "return true if a password matches a previous hardcoded password" in new Context {
+      val passwordInfo = PasswordInfo("bcrypt", "$2a$10$PBwXy.iQz9n4QOdbgEV7Ve2aYsvXeAvyT0rzhoZKwaDH/3j3tUSW.")
+
+      hasher.matches(passwordInfo, password) must beTrue
+    }
+
+    "return false if a password doesn't match a previous hashed password" in new Context {
+      val passwordInfo = hasher.hash(password)
+
+      hasher.matches(passwordInfo, "not-equal") must beFalse
+    }
+  }
+
+  "The `isSuitable` method" should {
+    "return true if the hasher is suitable for the given password info" in new Context {
+      val passwordInfo = hasher.hash(password)
 
-      hasher.matches(info, password) must beTrue
+      hasher.isSuitable(passwordInfo) must beTrue
     }
 
-    "return false if a password doesn't match a previous hashed password" in {
-      val password = "my_S3cr3t_p@sswQrd"
-      val hasher = new BCryptPasswordHasher
-      val info = hasher.hash(password)
+    "return false if the hasher isn't suitable for the given password info" in new Context {
+      val passwordInfo = PasswordInfo("scrypt", "")
 
-      hasher.matches(info, "not-equal") must beFalse
+      hasher.isSuitable(passwordInfo) must beFalse
     }
   }
 
-  "The hasher" should {
-    "be equal if the rounds parameter is equal" in {
-      val hasher1 = new BCryptPasswordHasher(1)
-      val hasher2 = new BCryptPasswordHasher(1)
+  "The `isDeprecated` method" should {
+    "return None if the hasher isn't suitable for the given password info" in new Context {
+      val passwordInfo = PasswordInfo("scrypt", "")
+
+      hasher.isDeprecated(passwordInfo) must beNone
+    }
+
+    "return Some(true) if the stored log rounds are not equal the hasher log rounds" in new Context {
```

Comment:
Add a similar test for `isSuitable`: "return true when given a password info from a bcrypt hasher with different log rounds"